### ISSUE TITLE
Use native-ssh with 'gcloud compute ssh'

### ISSUE THREAD

**Issue by MikaelSmith** [puppetlabs/bolt#3069]
## Describe the Bug
I'd like to use native-ssh support to run commands using `gcloud compute ssh`. The `gcloud` command supports several authentication methods that would be difficult to implement otherwise (like Identity-Aware Proxy).

When I try to do that I run into
```
Started on michael-ssh-target...
Failed on michael-ssh-target:
  Failed to connect to michael-ssh-target: ERROR: (gcloud.compute.ssh) unrecognized arguments:
    -o
    michaelsmith@michael-ssh-target
    exit
    To search the help text of gcloud commands, run:
    gcloud help -- SEARCH_TERMS
Failed on 1 target: michael-ssh-target
Ran on 1 target in 1.82 sec
```

## Expected Behavior
I would expect the configuration above to work, or some configuration to make it work.

I could probably make it work by creating a wrapper around the command, but I think there are a few clean improvements that can be made to how the command is constructed to make it more portable.

## Steps to Reproduce
1. Create an inventory with an SSH section like
    ```
        ssh:
          native-ssh: true
          ssh-command: [gcloud, compute, ssh]
          copy-command: [gcloud, compute, scp, --recurse]
    ```
2. Run `bolt command run whoami -t <target>`

 - Version: main branch
 - Platform: macOS 12.2.1

**Comment by MikaelSmith** [comment]
Hmm, my thinking was that I could separate the SSH command and options from commands to execute with `--`, but that's command-specific and may not be supported by all SSH replacements.

**Comment by beechtom** [comment]
Closed by #3070
